aspects/aspects.go

```diff
@@ -306,57 +328,144 @@ func (a *Aspect) Set(databag DataBag, request string, value interface{}) error {
 		return a.bundle.schema.Validate(data)
 	}
 
-	return &NotFoundError{
-		Account:    a.bundle.Account,
-		BundleName: a.bundle.Name,
-		Aspect:     a.Name,
-		Field:      request,
-		Cause:      "field not found",
+	return notFoundErrorFrom(a, request, "no matching write rule")
+}
+
+// Get returns the aspect value identified by the request. If either the named
+// aspect or the corresponding value can't be found, a NotFoundError is returned.
+func (a *Aspect) Get(databag DataBag, request string, value *interface{}) error {
+	matches, err := a.matchGetRequest(request)
+	if err != nil {
+		return err
+	}
+
+	var merged interface{}
+	for _, match := range matches {
+		var val interface{}
+		if err := databag.Get(match.storagePath, &val); err != nil {
+			if errors.Is(err, PathNotFoundError("")) {
+				continue
+			}
+			return err
+		}
+
+		// build a namespace based on the unmatched suffix parts
+		namespace := match.suffixParts
+		for i := len(namespace) - 1; i >= 0; i-- {
+			val = map[string]interface{}{namespace[i]: val}
+		}
+
+		// merge result with results from other matching rules
+		merged, err = merge(merged, val)
+		if err != nil {
+			return err
+		}
+	}
+
+	if merged == nil {
+		return notFoundErrorFrom(a, request, "no value for matching rules")
+	}
+
+	// the top level maps the request to the remaining namespace
+	*value = map[string]interface{}{request: merged}
```

Comment:
Now that `Get` only ever puts a `map[string]interface{}` inside of `value`, is there a reason not to just return a `map[string]interface{}` from this method, rather than making the caller pass in a pointer to an interface?

Reply:
Last time this was discussed I think it the intent was to keep it as close to the old config code as possible. Not sure, if that's still as relevant @pedronis